cognite/neat/rules/models/rules.py

```diff
@@ -330,6 +310,29 @@ def is_suffix_compliant(cls, value, values):
         else:
             return value
 
+    @validator("namespace", always=True)
+    @skip_field_validator("validators_to_skip")
+    def set_namespace_if_none(cls, value, values):
+        if value is None:
+            if values["prefix"] == values["suffix"]:
+                return Namespace(f"http://purl.org/cognite/{values['prefix']}#")
+            else:
+                return Namespace(f"http://purl.org/cognite/{values['prefix']}/{values['suffix']}#")
+        try:
+            return Namespace(parse_obj_as(HttpUrl, value))
```

Comment:
`parse_obj_as` is from `pydantic` `v1` lets use a `TypeAdapter`: https://docs.pydantic.dev/latest/migration/#introduction-of-typeadapter